### PR TITLE
Setup database testing

### PR DESCRIPTION
**Pull request by abeforgit**
Adds an abstract class to derive from if you want to test the DAOs

### CODE REVIEW

**Comment by abeforgit** [comment]
Mostly based on how [dropwizard tests their jdbi](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-jdbi3/src/test/java/io/dropwizard/jdbi3/JdbiTest.java

File: src/test/java/telraam/DatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import io.dropwizard.jersey.validation.Validators;
+import io.dropwizard.logging.BootstrapLogging;
+import io.dropwizard.setup.Environment;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.flywaydb.core.Flyway;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+public abstract class DatabaseTest {
```
Comment (abeforgit) [inline src/test/java/telraam/DatabaseTest.java]: needs javadoc and other comments

**Comment by mcbloch** [comment]
You included testConfig.yml twice, is this a mistake or intentional?
And if the latter, why?

**Comment by abeforgit** [comment]
I came to the point where I was basically re-implementing the entire bootstrap procedure, so I replaced everything with the AppExtension which gives you a full app.
There is one unfortunate type cast in it, but it should be a lot more clear what is going on now, and it is actually using intended features.

**Comment by mcbloch** [comment]
In the BatonDAOTest class you have this line `batonDAO = jdbi.onDemand(BatonDAO.class);
` twice. Both in the setup and in the createBaton test.

**Pull request merged**